core/unix/module_elf.c

```diff
@@ -756,6 +756,16 @@ get_proc_address_ex(module_base_t lib, const char *name, bool *is_indirect_code
     os_get_module_info_lock();
     ma = module_pc_lookup((app_pc)lib);
     if (ma != NULL) {
+        /* Check whether os_module_data_t was initialized during
```

Comment:
Described problem in details in #5946.
Apparently this is the same issue as #3385: dynamic section not initialized because of failed soname check during first flat-mmap point. The problem occurs when loading libraries with .dynstr is in its last +rw segment.